tensorflow/stream_executor/cuda/cuda_dnn.cc

```diff
@@ -6009,6 +6009,64 @@ bool CudnnSupport::DoActivate(Stream* stream,
   return IsStatusOk(status, /*report_error=*/true);
 }
 
+namespace {
+// Cudnn legacy API only supports int32 indexing and can handle a maximum of
+// 2^31-1 elements. For pooling operations, we split the big tensor along the
+// batch axis if necessary and call cudnn API sequentially.
+// This function will return a vector of tuples, representing how to split the
+// tensors: (batch number of each split, offset in bytes of the input, offset in
+// bytes of the output).
+port::StatusOr<std::vector<std::tuple<int64_t, int64_t, int64_t>>>
```

Comment:
Prefer structs over tuples. See [style guide](https://google.github.io/styleguide/cppguide.html#Structs_vs._Tuples)

Reply:
Thanks. I have pushed new commit to resolve this. PTAL.

Reply:
Thanks, Kaixi! Reads much better!